src/codec/data_storage/data_block.rs

```diff
@@ -0,0 +1,276 @@
+use elliptic_curve::rand_core::CryptoRngCore;
+use futures::{AsyncWrite, AsyncWriteExt};
+use winnow::binary::le_u8;
+use winnow::token::literal;
+use winnow::Parser;
+
+use super::data_options::DataOptions;
+use super::encrypted_data_chunk::EncryptedDataChunk;
+use crate::codec::crypto::{AuthenticationTag, Nonce};
+use crate::codec::header::BANYAN_DATA_MAGIC;
+use crate::codec::meta::{BlockSize, BlockSizeError};
+use crate::codec::{Cid, ParserResult, Stream};
+use crate::utils::std_io_err;
+
+use std::sync::{Arc, RwLock};
+
+pub struct DataBlock {
+    data_options: DataOptions,
+    cid: Arc<RwLock<Option<Cid>>>,
+    contents: Vec<EncryptedDataChunk>,
+}
+
+impl DataBlock {
+    /// The amount of payload a small encrypted block can hold.
+    #[allow(clippy::identity_op)]
+    pub const SMALL_ENCRYPTED_SIZE: usize =
+        262_144 - (1 * Nonce::size() + AuthenticationTag::size());
+
+    /// The amount of payload a standard encrypted block can hold.
+    pub const STANDARD_ENCRYPTED_SIZE: usize =
+        134_217_728 - (64 * Nonce::size() + AuthenticationTag::size());
```

Comment:
I thought I was going to have to modify these to account for not including a signature anymore but I'm a little confused on what they are encoding right now. Each chunk needs a nonce and an Auth tag, so it looks like the standard one should actually read: ` 134_217_728 - 64*( Nonce::size() + AuthenticationTag::size());` unless I am misunderstanding. Additionally they don't seems to account for the CID table or the signature. 

I just want to make sure I understand if this is incorrect if I am misunderstanding something before I change anything

Reply:
So those values were intended to be the amount of data that could be stored in block while keeping the overall size of the data block the same. I'm not sure it quite hits the mark though. Each block size has two important numbers the total size (n) and the chunk size (m) defined by an exponent of 2. The exponents of two here allow us to guarantee whatever number of chunks are selected the total size will be able to be evenly divided into that many chunks.

For our small blocks n = 18, and m = 18, and our standard ones are n = 26, and m = 20. Ideally the final written-to-disk size of a data block would be the size of the fixed header + 2^n bytes + an optional fixed trailer. Currently the signature lives in that fixed trailer so is not included in those counts. 

To keep the size of a chunk the same whether its encrypted or not, we need to reduce the storable data size in each chunk by the encryption overhead which is the Nonce and AuthenticationTag. So what these sizes are broken down into is:

$$TOTAL_CHUNK_SIZE = 2^n$$

$$TOTAL_CHUNK_COUNT = 2^n / 2^m$$

$$PER_CHUNK_DATA = 2^m - (NONCE_SIZE + AUTHENTICATION_TAG_SIZE)$$

Thus the total storable data is:

$$TOTAL_STORABLE = PER_CHUNK_DATA * TOTAL_CHUNK_COUNT$$

Our small data blocks only have a single chunk, while our standard ones have 64 (thus the hardcoded multiple).

Reply:
So yeah these ones don't need to currently be modified if the signature is removed, but they will need to be modified if we decide we do ultimately want a per-chunk signature. I don't think those are necessary though.

Reply:
One thing probably worth checking, I don't think this is account for the included length of the data... Which means we might have an off-by-a-couple-bytes error in this size...